_data-prepper/pipelines/configuration/processors/otel-trace-group.md

```diff
@@ -0,0 +1,79 @@
+---
+layout: default
+title: OTel trace group
+parent: Processors
+grand_parent: Pipelines
+nav_order: 45
+---
+
+# OTel trace group processor
+
+The OpenTelemetry (OTel) trace group processor completes missing trace group related fields in the collection of [span](https://github.com/opensearch-project/data-prepper/blob/834f28fdf1df6d42a6666e91e6407474b88e7ec6/data-prepper-api/src/main/java/org/opensearch/dataprepper/model/trace/Span.java) records by looking up the OpenSearch backend. The OTel trace group processor identifies the missing trace group information for a `spanId` by looking up the relevant fields in its root `span` stored in OpenSearch.
+
+## OpenSearch
+
+See the following example `YAML` configuration file for the OTel trace group processor when you connect to an OpenSearch cluster using your username and password:
+
+``` YAML
+pipeline:
+  ...
+  processor:
+    - otel_trace_group:
+        hosts: ["https://localhost:9200"]
+        cert: path/to/cert
+        username: YOUR_USERNAME_HERE
+        password: YOUR_PASSWORD_HERE
+```
+
+See [OpenSearch security](https://github.com/opensearch-project/data-prepper/blob/834f28fdf1df6d42a6666e91e6407474b88e7ec6/data-prepper-plugins/opensearch/opensearch_security.md#L4) for a more detailed explanation of which OpenSearch credentials and permissions are required and how to configure those credentials for the OTel trace group processor.
+
+### Amazon OpenSearch Service
+
+See the following example `YAML` configuration file for the OTel trace group processor when you use the Amazon OpenSearch Service:
+
+``` YAML
+pipeline:
+  ...
+  processor:
+    - otel_trace_group:
+        hosts: ["https://your-amazon-opensearch-service-endpoint"]
+        aws_sigv4: true
+        cert: path/to/cert
+        insecure: false
+```
+
+## Configuration
+
+You can configure the OTel trace group processor with the following options.
+
+| Name | Description | Default value |
+| -----| ----| -----------|
+| `hosts`| A list of IP addresses of OpenSearch nodes. Required. | No default value. | 
+| `cert` | A certificate authority (CA) certificate that is PEM-encoded. Accepts both .pem or .crt. This enables the client to trust the CA that has signed the certificate that OpenSearch is using. Optional. | `null` |
+| `aws_sigv4` | A Boolean flag to sign the HTTP request with AWS credentials. Only applies to Amazon OpenSearch Service. See [OpenSearch security](https://github.com/opensearch-project/data-prepper/blob/129524227779ee35a327c27c3098d550d7256df1/data-prepper-plugins/opensearch/security.md) for details. Optional. | `false`. |
+| `aws_region` | A string that represents the region of Amazon OpenSearch Service domain, for example, `us-west-2`. Only applies to Amazon OpenSearch Service. Optional. | `us-east-1` |
```

Comment:
"The Region of the Amazon OpenSearch Service domain" (we actually capitalize "Region" here)